src/tcgwars/logic/impl/gen3/DeltaSpecies.groovy

```diff
@@ -3119,21 +3151,20 @@ public enum DeltaSpecies implements LogicCardInfo {
         }
       };
       case METAGROSS_STAR_113:
-      return basic (this, hp:HP070, type:M, retreatCost:3) {
+      return basic (this, hp:HP090, type:M, retreatCost:3) {
         weakness R
         resistance G, MINUS30
         move "Critical Collection", {
           text "10 damage. Count the number of Prize cards your opponent has taken. Search your discard pile for up to that many [M] Energy cards and attach them to Metagross Star."
           energyCost M
           onAttack {
             damage 10
-
-            def takenPrizes = opp.prizeCardSet.takenCount
-            def energies = my.discard.filterByType(ENERGY).size()
-            if (takenPrizes && energies) {
-              def count = Math.min(takenPrizes, energies)
-              (1..count).each {
-                attachEnergyFrom(type:M, my.discard, self)
+            afterDamage{
+              def takenPrizes = opp.prizeCardSet.takenCount
+              def energies = my.discard.filterByType(ENERGY).size()
```

Comment:
How about `filterByType(ENERGY).filterByEnergyType(M).size()`?

Reply:
Good catch, will commit in a bit 👍